jsonnet/kube-prometheus/kube-prometheus.libsonnet

```diff
@@ -50,7 +50,62 @@ local configMapList = k3.core.v1.configMapList;
         preserveUnknownFields: null,
       }),
     },
-  },
+    service+: {
+      spec+: {
+        ports: [
+          {
+            name: 'https',
+            port: 8443,
+            targetPort: 'https',
+          },
+        ],
+      },
+    },
+    serviceMonitor+: {
+      spec+: {
+        endpoints: [
+          {
+            port: 'https',
+            scheme: 'https',
+            honorLabels: true,
+            bearerTokenFile: '/var/run/secrets/kubernetes.io/serviceaccount/token',
+            tlsConfig: {
+              insecureSkipVerify: true,
```

Comment:
Nit: Could we have a section in the docs on saying to users to make this false when they have a cert, wdyt?

Reply:
this is actually a good one :thinking: I would even go further and:
1. make it per default `false` and be it a variable in `$._config`.
2. add cert setting variables in `$._config`

Reply:
I used the same mechanism as we use in other places (mostly in KSM). From what I remember in the past we didn't want to do anything more, if that changed let's open an issue and discuss it there as this is something that should be done in all places at the same time, WDYT?

Reply:
hack day project idea: example setup with cert-manager, showing how to not skip tls verification

Reply:
Let's discuss it further in https://github.com/coreos/kube-prometheus/issues/469